lib/puma/reactor.rb

```diff
@@ -122,36 +129,48 @@ def initialize(server, app_pool)
     # This calculation happens in `calculate_sleep`.
     def run_internal
       sockets = @sockets
+      selector = @selector
 
       while true
         begin
-          ready = IO.select sockets, nil, nil, @sleep_for
+          ready = selector.select @sleep_for
         rescue IOError => e
           Thread.current.purge_interrupt_queue if Thread.current.respond_to? :purge_interrupt_queue
-          if sockets.any? { |socket| socket.closed? }
+          if sockets.any? { |socket| m.value.closed? }
```

Comment:
`m.value`?

Reply:
You've used this all over the place put I'm totally confused on where `m` comes from.